plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/AppEngineStandardWizardPage.java

```diff
@@ -64,23 +66,37 @@ public void createControl(Composite parent) {
     PlatformUI.getWorkbench().getHelpSystem().setHelp(container,
         "com.google.cloud.tools.eclipse.appengine.newproject.NewProjectContext"); //$NON-NLS-1$
 
-    ModifyListener pageValidator = new PageValidator();
 
-    // Java package name
-    Label packageNameLabel = new Label(container, SWT.NONE);
-    packageNameLabel.setText(Messages.getString("java.package")); //$NON-NLS-1$
-    javaPackageField = new Text(container, SWT.BORDER);
-    GridData javaPackagePosition = new GridData(GridData.FILL_HORIZONTAL);
-    javaPackagePosition.horizontalSpan = 2;
-    javaPackageField.setLayoutData(javaPackagePosition);
-    javaPackageField.addModifyListener(pageValidator);
+    createPackageField(container);
 
     // Manage APIs
     appEngineLibrariesSelectorGroup = new AppEngineLibrariesSelectorGroup(container);
 
     Dialog.applyDialogFont(container);
   }
 
+  // Java package name
+  private void createPackageField(Composite container) {
+    
+    Composite composite = new Composite(container, SWT.NONE);
+    GridData compositePosition = new GridData(GridData.FILL_HORIZONTAL);
+    compositePosition.horizontalSpan = 2;
+    composite.setLayoutData(compositePosition);
```

Comment:
OK, I think I see what you're saying. It's container that's the issue, not composite. 

In this case container is supplied by the environment, so it doesn't really matter which method we do it in or which method we create the composite in; but we can explicitly set a gridLayout on it. How's this?